plugins/modules/gitlab_merge_request.py

```diff
@@ -0,0 +1,414 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2023, Ondrej Zvara (ozvara1@gmail.com)
+# Based on code:
+# Copyright (c) 2021, Lennert Mertens (lennert@nubera.be)
+# Copyright (c) 2021, Werner Dijkerman (ikben@werner-dijkerman.nl)
+# Copyright (c) 2015, Werner Dijkerman (ikben@werner-dijkerman.nl)
+# Copyright (c) 2019, Guillaume Martinez (lunik@tiwabbit.fr)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+module: gitlab_merge_request
+short_description: Create, update, or delete GitLab merge requests
+version_added: 7.1.0
+description:
+  - Creates a merge request if it does not exist.
+  - When a merge request does exist, it will be updated if the provided parameters are different.
+  - When a merge request does exist and I(state=absent), the merge request will be deleted.
+  - Existing merge requests are matched based on I(title), I(source_branch), I(target_branch),
+    and I(state_filter) filters.
+author:
+  - zvaraondrej (@zvaraondrej)
+requirements:
+  - Python >= 2.7
+  - python-gitlab >= 2.3.0
+extends_documentation_fragment:
+  - community.general.auth_basic
+  - community.general.gitlab
+  - community.general.attributes
+
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+
+options:
+  state:
+    description:
+      - Create or delete merge request.
+    default: present
+    type: str
+    choices: ["present", "absent"]
+  project:
+    description:
+      - The path or name of the project.
+    required: true
+    type: str
+  source_branch:
+    description:
+      - Merge request's source branch.
+      - Ignored while updating existing merge request.
+    required: true
+    type: str
+  target_branch:
+    description:
+      - Merge request's target branch.
+    required: true
+    type: str
+  title:
+    description:
+      - A title for the merge request.
+    type: str
+    required: true
+  description:
+    description:
+      - A description for the merge request.
+      - Gets overriden by a content of file specified at I(description_path), if found.
```

Comment:
Now that the `main` and `stable-7` branches are switching to [semantic markup](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#semantic-markup-within-module-documentation), let's use it here as well:
```suggestion
      - Gets overriden by a content of file specified at O(description_path), if found.
```
(This allows the docsite to put a link here to the `description_path` option.)